src/query/block/meta.go

```diff
@@ -43,3 +48,82 @@ func (m Metadata) Equals(other Metadata) bool {
 func (m Metadata) String() string {
 	return fmt.Sprintf("Bounds: %v, Tags: %v", m.Bounds, m.Tags)
 }
+
+// ResultMetadata describes metadata common to each type of query results,
+// indicating any additional information about the result.
+type ResultMetadata struct {
+	// LocalOnly indicates that this query was executed only on the local store.
+	LocalOnly bool
+	// Exhaustive indicates whether the underlying data set presents a full
+	// collection of retrieved data.
+	Exhaustive bool
+	// Warnings is a list of warnings that indicate potetitally partial or
+	// incomplete results.
+	Warnings []Warning
+}
+
+// NewResultMetadata creates a new result metadata.
+// NB: capacity for warning is initialized to 1, since it is unlikely that more
+// than one warning will exist.
```

Comment:
Maybe worth even just leaving it nil?  Only because I assume most queries won't cause a warning, and it will unfortunately almost definitely be a heap allocation (since it's a slice and pointer type).

Reply:
Good call